### PR TITLE
plot: split Ticker into Ticker and Labeler

### PR DESCRIPTION
**Pull request by kostya-sh**
DO NOT MERGE.

This change is for review and discussion only.

This is backward incompatible change.

### CODE REVIEW

**Comment by kortschak** [comment]
I like this.

File: axis.go [hunk 1]
```diff
@@ -451,76 +468,100 @@ func (ts ConstantTicks) Ticks(float64, float64) []Tick {
...
 }
 
-var _ Ticker = UnixTimeTicks{}
+var _ Labeler = UnixTimeLabeler{}
 
 // Ticks implements plot.Ticker.
-func (utt UnixTimeTicks) Ticks(min, max float64) []Tick {
-	if utt.Ticker == nil {
-		utt.Ticker = DefaultTicks{}
+func (utt UnixTimeLabeler) Labels(ticks []Tick) []string {
```
Comment (sbinet) [inline axis.go]: perhaps `s/utt/utl/g` ?

File: axis.go [hunk 2]
```diff
@@ -24,6 +24,11 @@ type Ticker interface {
 	Ticks(min, max float64) []Tick
 }
 
+// Labeler creates lables from Ticks
```
Comment (sbinet) [inline axis.go]: `s/lables/labels/`

**Comment by sbinet** [comment]
looks good.
I may even abandon #278 in favor of this change.

**Comment by kostya-sh** [comment]
If the agreement is that the new API is an improvement over the current API and will be merged I am happy to finsh this PR (it requires some cleanup, more docs and tests).

I also found previous discussion about tick generation (https://groups.google.com/forum/#!topic/gonum-dev/g7ZgAVItbbA) and https://github.com/vdobler/plot-1/blob/redesign/design.md by @vdobler. There are a few proposed features. Maybe it is worth checking how to implement them (and if it is possible at all) with the new API?

**Comment by sbinet** [comment]
(apologies for the belated answer)

yes, I guess it would be worth to get a sense of how this might look like.
(I myself have still to update my PR #245 and/or provide a more theme-based oriented facility, like https://godoc.org/golang.org/x/exp/shiny/widget/theme#Theme)

**Comment by eaburns** [comment]
Any news on this? We are trying to clean up our PRs. Folks seemed to like this idea, but it still has some outstanding comments. Do you plan to address them? If not, we plan to close this soon.

**Comment by kostya-sh** [comment]
I am going to close this PR. I am still not sure if the proposed interface is flexible enough. Additionally I do not have much time to work on this at the moment.

**Pull request closed without merging**